### PR TITLE
Get Api file tested task 3

### PR DESCRIPTION
**Pull request by oaissnk**

### CODE REVIEW

File: db/app.controller.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { selectAllTopics } = require("./app.models");
+exports.getAPI = (_, res) => {
+  res.status(200).send({ message: "all ok" });
+};
+
```
Comment (Dutton1990) [inline db/app.controller.js:5]: Is this what task 3 is asking for with the endpoint /api?

File: db/app.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const app = express();
+const { handlePsqlErrors, handleCustomErrors, handleServerErrors } = require("./error");
+const { getAPI, getAllTopics, routeNotFound } = require("./app.controller");
+app.use(express.json());
```
Comment (Dutton1990) [inline db/app.js:5]: Do you need express.json yet? Only push code that is relevant to the ticket please!

File: db/app.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+app.get("/api", getAPI);
+module.exports = app;
+
+app.get('/api/topics', getAllTopics)
+
+app.all("*", routeNotFound)
```
Comment (Dutton1990) [inline db/app.js:11]: Nice implementation for a bad path!

File: db/app.models.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const db = require("./index");
+
+exports.selectAllTopics = () => {
+  return db.query(`SELECT * FROM topics;`).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({
+        status: 404,
+        message: `No path found`,
+      });
+    }
```
Comment (Dutton1990) [inline db/app.models.js:10]: I would experiment and put a console log here and see if you ever get in this if block - it's not the reason for your sad path test passing.

**Review by Dutton1990 — COMMENTED** [review]
Hi Oaiss 👋 

I'm unsure which ticket I'm looking at. 

If this is for task 2 - GET /api/topics - then this looks good to me and you can merge. 

However your title says it's for task 3 - GET /api. If that is the case then this ticket doesn't do what the task says. Please reattempt this ticket and resubmit the PR for one of us to check 👍

File: db/app.js (hunk 3 above)
Comment (oaissnk) [reply db/app.js:11]: thanks :)

**Pull request closed without merging**